### ISSUE TITLE
Race condition with async and HTML import dedupes

### ISSUE THREAD

**Issue by panuhorsmalahti** [Polymer/polymer#3095]
I have a component X which depends on component Y (my-socketio). I'm loading it without async. In certain cases the component Y is loaded _after_ component X begins to register itself. I finally tracked this issue down, and I can replicate it (for about 50% of the page refreshes). It doesn't seem to occur on Firefox.

I'm using Jade syntax, I can translate to HTML later if necessary.

Component X:

```
link(rel="import", href="../../lib/polymer/polymer.html")
link(rel="import", href="../../components/my-socket-io/my-socket-io.html")
script(src="../../../components/my-get/my-get.min.js")
```

When the issue occurs, my-socket-io's .js file is executed _after_ my-get.min.js.

The issue happens when there's a completely different component dependency tree like this:
Component A -> Component B -> Component Y.

Here Component A imports Component B asynchronously, but Component Y synchronously! (Although I'd consider it bug even if async was used in both of the dependencies).

If I change the Component A import of Component B to be synchronous, the issue disappears.

I believe this is a bug, because Component X shouldn't be broken due to sync or async imports of Component Y elsewhere in the system.

I don't have right now to get a runnable test case, hopefully I can do that later.

**Comment by ebidel** [comment]
The difference is between FF and Chrome is native imports vs polyfill'd imports. Can you post a jsbin that repos the problem? It's a little hard to follow all the A,B,X's and what is and is not async.

**Comment by ebidel** [comment]
Related to https://github.com/Polymer/polymer/issues/2522?

**Comment by hickscorp** [comment]
@panuhorsmalahti Have a look at #3118. Maybe my solution will help you fix this problem.

**Comment by aigan** [comment]
I submitted #4443 as a consistent example of probably the same bug

**Comment by TimvdLippe** [comment]
Closing this issue as an upstream Chrome issue filed at https://bugs.chromium.org/p/chromium/issues/detail?id=669710
